heartbeat/docs/reference/configuration/heartbeat-options.asciidoc

```diff
@@ -0,0 +1,448 @@
+payload is sent, but the client expects to receive a payload on connect.
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: tcp 
+  schedule: '@every 5s'
+  hosts: ["myhost"]
+  ports: [80, 9200, 5044]
+  ipv4: true
+  ipv6: true
+  mode: any
+  timeout: 16s
+  check:
+    send: ''
+    receive: ''
+-------------------------------------------------------------------------------
+
+
+//Steffen: Can you provide a good example of the values to set for send and receive in the above example?
+
+[[monitor-tcp-proxy-url]]
+===== proxy_url
+
+The SOCKS5 proxy URL.
+
+//Steffen: Can we say more about this option?
+
+[[monitor-tcp-proxy-use-local-resolver]]
+===== proxy_use_local_resolver
+
+A Boolean value that specifies whether to resolve hostnames locally instead of
+resolving them on the SOCKS5 server. The default is false.
+
+[[monitor-tcp-tls-ssl]]
+===== ssl
+
+The TLS/SSL connection settings. 
+
+//Steffen: Is this option so that you can check to see if the servers certificates are valid?
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: tcp 
+  schedule: '@every 5s'
+  hosts: ["myhost"]
+  ports: [80, 9200, 5044]
+  ipv4: true
+  ipv6: true
+  mode: any
+  timeout: 16s
+  ssl:
+    certificate_authorities: ['/etc/ca.crt']
+    supported_protocols: ["TLSv1.0", "TLSv1.1", "TLSv1.2"]
+-------------------------------------------------------------------------------
+
+
+//Steffen: How are the values for supported_protocols used?
+
+[[monitor-http-options]]
+==== HTTP Options
+
+These options are valid when the `type` is `http`.
+
+[[monitor-http-urls]]
+===== urls
+
+A list of URLs to ping.
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: http 
+  schedule: '@every 5s'
+  urls: ["http://myhost:80"]
+  ipv4: true
+  ipv6: true
+  mode: any
+-------------------------------------------------------------------------------
+
+
+[[monitor-http-proxy-url]]
+===== proxy_url
+
+The HTTP proxy URL. This setting is optional. If not set, the HTTP_PROXY
+environment variable is used.
+
+[[monitor-http-username]]
+===== username
+
+The username for authenticating with the server. This setting is optional. 
+
+[[monitor-http-password]]
+===== password
+
+The password for authenticating with the server. This setting is optional.
+
+//Steffen: When are authentication credential required? If you are sending a request and expecting a response?
+
+[[monitor-http-tls-ssl]]
+===== ssl
+
+The TLS/SSL connection settings for use with the HTTPS endpoint. If you don't
+specify settings, the system defaults are used. 
+
+
+Example configuration:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+- type: http 
+  schedule: '@every 5s'
+  urls: ["http://myhost:80"]
```

Comment:
@urso Do you want the example changed? I show the ssl options in a later example.